modules/proxy.js

```diff
@@ -0,0 +1,131 @@
+'use strict';
+const rp = require('request-promise');
+const absolution = require('absolution');
+const jsdom = require('jsdom');
+const { JSDOM } = jsdom;
+const config = require('config');
+
+const legacyUrl = config.get('legacyDomain');
+
+const postToLegacyForm = (req, res) => {
+    res.cacheControl = { maxAge: 0 };
+
+    // work out if we need to serve english/welsh page
+    let localePath = req.i18n.getLocale() === 'cy' ? config.get('i18n.urlPrefix.cy') : '';
+    let pagePath = localePath + req.path;
+
+    return rp
+        .post({
+            uri: legacyUrl + pagePath,
+            form: req.body,
+            strictSSL: false,
+            jar: true,
+            simple: true,
+            followRedirect: false,
+            resolveWithFullResponse: true,
+            followOriginalHttpMethod: true
+        })
+        .catch(err => {
+            const proxyResponse = err.response;
+            if (proxyResponse.statusCode === 302) {
+                res.redirect(302, proxyResponse.headers.location);
+            } else {
+                // @TODO can we send them somewhere better?
+                res.redirect('/home');
+            }
+        });
+};
+
+const proxyLegacyPage = (req, res, domModifications, pathOverride) => {
+    res.cacheControl = { maxAge: 0 };
+
+    // work out if we need to serve english/welsh page
+    let localePath = req.i18n.getLocale() === 'cy' ? config.get('i18n.urlPrefix.cy') : '';
+
+    // allow a custom path (eg. to serve / over /legacy)
+    // which would otherwise fail
+    let pagePath = pathOverride ? pathOverride : localePath + req.path;
+
+    return rp({
+        url: legacyUrl + pagePath,
+        qs: req.query,
+        strictSSL: false,
+        jar: true,
+        resolveWithFullResponse: true
+    })
+        .then(response => {
+            let body = response.body;
+            // convert all links in the document to be absolute
+            // (only really useful on non-prod envs)
+            body = absolution(body, 'https://www.biglotteryfund.org.uk');
+
+            // fix meta tags in HTML which use the wrong CNAME
+            body = body.replace(/wwwlegacy/g, 'www');
+
+            // parse the DOM
+            let dom = new JSDOM(body);
+
+            if (req.app.get('env') === 'development') {
+                let titleText = dom.window.document.title;
+                dom.window.document.title = '[PROXIED] ' + titleText;
```

Comment:
Not sure what you think about leaving this in @davidrapson, it was solely to make it easier to eyeball a page that's been modified/proxied, as opposed to redirected/external.

Reply:
Yep, I like this (it's scoped to development envs right?)